test/widgets/message_list_test.dart

```diff
@@ -620,6 +621,93 @@ void main() {
     });
   });
 
+  group('Update Narrow on message move', () {
+    final channel = eg.stream(name: 'move test stream');
+    final otherChannel = eg.stream(name: 'other move test stream');
+    final narrow = TopicNarrow(channel.streamId, 'example topic');
+
+    void prepareGetMessageResponse(List<Message> messages) {
+      connection.prepare(json: newestResult(
+        foundOldest: false, messages: messages).toJson());
+    }
+
+    void handleMessageMoveEvent(List<StreamMessage> messages, String newTopic, {int? newChannelId}) {
+      store.handleEvent(eg.updateMessageEventMoveFrom(
+        origMessages: messages,
+        newTopic: newTopic,
+        newStreamId: newChannelId,
+        propagateMode: PropagateMode.changeAll));
+    }
+
+    testWidgets('compose box send message after move', (WidgetTester tester) async {
+      final message = eg.streamMessage(stream: channel, content: 'Message to move');
+      await setupMessageListPage(tester, narrow: narrow, messages: [message], streams: [channel, otherChannel]);
+
+      final channelContentInputFinder = find.descendant(
+        of: find.byType(ComposeAutocomplete),
+        matching: find.byType(TextField));
+
+      await tester.enterText(channelContentInputFinder, 'Some text');
+      check(tester.widget<TextField>(channelContentInputFinder))
+        ..decoration.isNotNull().hintText.equals('Message #${channel.name} > example topic')
+        ..controller.isNotNull().text.equals('Some text');
+      prepareGetMessageResponse([message]);
```

Comment:
nit:
```suggestion
        ..controller.isNotNull().text.equals('Some text');

      prepareGetMessageResponse([message]);
```

Generally in a test case when there's a sequence like
* make something happen or change
* check the state of the world
* make something else happen or change
* check the new state of the world
* etc.

I find it helpful, for the reader, to group those change/check pairs into stanzas using blank lines.